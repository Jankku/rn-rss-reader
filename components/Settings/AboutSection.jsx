import { View, Text, SectionList, StyleSheet } from 'react-native';
import { applicationName, nativeApplicationVersion } from 'expo-application';
import { openBrowserAsync } from 'expo-web-browser';
import { Ionicons } from '@expo/vector-icons';
import { useTheme } from '@react-navigation/native';
import SettingsSectionItem from './SettingsSectionItem';
import SettingsSectionHeader from './SettingsSectionHeader';
import ItemDivider from '../NewsFeed/ItemDivider';

const aboutSectionData = [
  {
    title: 'About',
    data: [
      {
        title: `${applicationName ?? '-'}`,
        value: `${nativeApplicationVersion ?? '-'}`,
        icon: 'logo-android',
      },
      {
        title: 'Developer',
        value: 'Jankku',
        link: 'https://github.com/Jankku/',
        icon: 'person',
      },
      {
        title: 'GitHub',
        value: 'https://github.com/Jankku/rn-rss-reader',
        link: 'https://github.com/Jankku/rn-rss-reader',
        icon: 'logo-github',
      },
    ],
  },
];

function AboutSection() {
  const { colors } = useTheme();
  const styles = makeStyles(colors);

  const openLink = (item) => {
    if (item.link) openBrowserAsync(item.link);
  };

  const _renderItem = ({ item }) => (
    <SettingsSectionItem onPress={() => openLink(item)}>
      <View style={styles.container}>
        <Ionicons name={item.icon} size={24} color={colors.text} style={styles.icon} />
        <View>
          <Text style={styles.title}>{item.title}</Text>
          <Text style={styles.value}>{item.value}</Text>
        </View>
      </View>
    </SettingsSectionItem>
  );

  const _renderSectionHeader = ({ section: { title } }) => <SettingsSectionHeader title={title} />;

  const _keyExtractor = (item, index) => item + index;

  return (
    <SectionList
      sections={aboutSectionData}
      renderSectionHeader={_renderSectionHeader}
      renderItem={_renderItem}
      keyExtractor={_keyExtractor}
      ItemSeparatorComponent={ItemDivider}
    />
  );
}

const makeStyles = (colors) =>
  StyleSheet.create({
    container: {
      alignContent: 'center',
      alignItems: 'center',
      flexDirection: 'row',
    },
    icon: {
      paddingEnd: 16,
    },
    title: {
      color: colors.text,
      fontSize: 16,
      fontWeight: '700',
    },
    value: {
      color: colors.text,
    },
  });

export default AboutSection;
