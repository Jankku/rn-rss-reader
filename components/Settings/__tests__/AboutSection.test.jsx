import AboutSection from '../AboutSection';
import { render } from '@testing-library/react-native';

describe('Test AboutSection', () => {
  test('Should match snapshot', () => {
    const { toJSON } = render(<AboutSection />);
    expect(toJSON()).toMatchSnapshot();
  });
});
