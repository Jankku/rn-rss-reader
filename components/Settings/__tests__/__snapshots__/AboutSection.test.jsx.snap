// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AboutSection Should match snapshot 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "icon": "logo-android",
            "title": "mock",
            "value": "mock",
          },
          Object {
            "icon": "person",
            "link": "https://github.com/Jankku/",
            "title": "Developer",
            "value": "Jankku",
          },
          Object {
            "icon": "logo-github",
            "link": "https://github.com/Jankku/rn-rss-reader",
            "title": "GitHub",
            "value": "https://github.com/Jankku/rn-rss-reader",
          },
        ],
        "title": "About",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontSize": 22,
            "fontWeight": "700",
            "paddingHorizontal": 16,
            "paddingVertical": 8,
          }
        }
      >
        About
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <View>
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 16,
                    "fontWeight": "700",
                  }
                }
              >
                mock
              </Text>
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                  }
                }
              >
                mock
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgb(216, 216, 216)",
              "height": 1,
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgb(255, 255, 255)",
              "padding": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <Text />
            <View>
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 16,
                    "fontWeight": "700",
                  }
                }
              >
                Developer
              </Text>
              <Text
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                  }
                }
              >
                Jankku
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgb(216, 216, 216)",
              "height": 1,
            }
          }
        />
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "padding": 16,
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text />
          <View>
            <Text
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontSize": 16,
                  "fontWeight": "700",
                }
              }
            >
              GitHub
            </Text>
            <Text
              style={
                Object {
                  "color": "rgb(28, 28, 30)",
                }
              }
            >
              https://github.com/Jankku/rn-rss-reader
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
